### PR TITLE
[DSM] [Playwright] PEPPER-1104 Upload Onc History test

### PR DESCRIPTION
**Pull request by aweng98**
[https://broadworkbench.atlassian.net/browse/PEPPER-1104](https://broadworkbench.atlassian.net/browse/PEPPER-1104)

Tests passed on Dev and Test env.

Screenshot 2023-10-23 at 10 17 58 AM

### CODE REVIEW

File: playwright-e2e/tests/dsm/onc-history/lms-onc-history-upload.spec.ts [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+    await test.step('Create mocked text file', async () => {
+      const rows = await participantListTable.rowsCount;
+      expect(rows).toBeGreaterThanOrEqual(1);
+      // Pick a random participant Short ID
+      const newArray = shuffle([...Array(rows).keys()]);
+      const rowIndex = newArray[0];
+      // RECORD_ID corresponds to Short ID in upload .txt file
+      // shortId = await participantListTable.getParticipantDataAt(rowIndex, 'Short ID');
+      // TODO Remove hardcode ID
+      shortId = 'PEX93B';
```
Comment (aweng98) [inline playwright-e2e/tests/dsm/onc-history/lms-onc-history-upload.spec.ts:82]: TODO: Undo hardcoded Short ID

File: playwright-e2e/utils/faker-utils.ts [hunk 2]
```diff
@@ -10,7 +10,11 @@ export const generateUserName = (namePrefix: string): string => {
   return `${namePrefix}-${faker.person.lastName()}${faker.lorem.word()}`;
 };
 
-export const generateRandomPhoneNum = (): string => Math.random().toString().slice(2, 11);
+// Generate US phone number (9 digits) with dashes
```
Comment (aweng98) [inline playwright-e2e/utils/faker-utils.ts:13]: TODO: Update comment: 10 digits

File: playwright-e2e/dsm/component/navigation/enums/miscellaneousNav-enum.ts [hunk 3]
```diff
@@ -1,9 +1,10 @@
 export enum MiscellaneousEnum {
-  MAILING_LIST = 'Mailing List',
-  PARTICIPANT_WITHDRAWAL = 'Participant Withdrawal',
-  PARTICIPANT_EVENT = 'Participant Event',
-  FOLLOW_UP_SURVEY = 'Follow-Up Survey',
```
Comment (aweng98) [inline playwright-e2e/dsm/component/navigation/enums/miscellaneousNav-enum.ts:5]: rearranged in alphabetical order.

File: playwright-e2e/data/upload/osteo2_onc_history_upload.txt [hunk 4]
```diff
@@ -0,0 +1,3 @@
+RECORD_ID	DATE_PX	TYPE_PX	LOCATION_PX	HISTOLOGY	ACCESSION	FACILITY	PHONE	FAX	DESTRUCTION	BLOCKS_WITH_TUMOR	TUMOR_SIZE	LOCAL_CONTROL	NECROSIS	VIABLE_TUMOR	FFPE	DECALCIFICATION	BLOCK_TO_REQUEST	REQUEST_STATUS
```
Comment (aweng98) [inline playwright-e2e/data/upload/osteo2_onc_history_upload.txt:1]: TODO: Delete, not used.

**Comment by Rinchan20** [comment]
got it to pass on dev

**Review by Rinchan20 — APPROVED** [review]
passed for me

**Pull request merged**